conans/client/build/cmake_toolchain_build_helper.py

```diff
@@ -0,0 +1,142 @@
+    args = []
+    compiler_version = conanfile.settings.get_safe("compiler.version")
+    if generator and parallel:
+        if ("Makefiles" in generator or "Ninja" in generator) and "NMake" not in generator:
+            args.append("-j%i" % cpu_count(conanfile.output))
+        elif "Visual Studio" in generator and compiler_version and Version(compiler_version) >= "10":
+            # Parallel for building projects in the solution
+            args.append("/m:%i" % cpu_count(output=conanfile.output))
+
+    if generator and msbuild_verbosity:
+        if "Visual Studio" in generator and compiler_version and Version(compiler_version) >= "10":
+            args.append("/verbosity:%s" % msbuild_verbosity)
+
+    return args
+
+
+class CMakeToolchainBuildHelper(object):
+    """ CMake helper to use together with the toolchain feature, it has the same interface
+        as the original 'conans.client.build.cmake.CMake' helper, but it will warn the
+        user about arguments forbidden, not used,... and how to achieve the same behavior
+        with the toolchain.
+    """
+
+    def __init__(self, conanfile, generator=None, build_folder=None, parallel=True,
+                 msbuild_verbosity="minimal"):
+
+        _validate_recipe(conanfile)
+
+        # assert generator is None, "'generator' is handled by the toolchain"
+        self._generator = generator or get_generator(conanfile)
+        self._is_multiconfiguration = is_multi_configuration(self._generator)
+
+        # Store a reference to useful data
+        self._conanfile = conanfile
+        self._parallel = parallel
+        self._msbuild_verbosity = msbuild_verbosity
+
+        self._build_folder = build_folder
+        self._cmake_program = "cmake"  # Path to CMake should be handled by environment
+
+    def configure(self, source_folder=None):
+        # TODO: environment?
+        if not self._conanfile.should_configure:
+            return
+
+        source = self._conanfile.source_folder
+        if source_folder:
+            source = os.path.join(self._conanfile.source_folder, source_folder)
+
+        build_folder = self._conanfile.build_folder
+        if self._build_folder:
+            build_folder = os.path.join(self._conanfile.build_folder, self._build_folder)
+
+        defs = {"CMAKE_TOOLCHAIN_FILE": CMakeToolchain.filename}
+
+        mkdir(build_folder)
+        arg_list = join_arguments([
+            defs_to_string(defs),
+            args_to_string([source])
+        ])
+        generator = '-G "{}" '.format(self._generator) if self._generator else ""
+        command = "%s %s%s" % (self._cmake_program, generator, arg_list)
+
+        is_windows_mingw = platform.system() == "Windows" and self._generator == "MinGW Makefiles"
+        self._conanfile.output.info("CMake command: %s" % command)
+        with chdir(build_folder):
+            if is_windows_mingw:
+                with tools.remove_from_path("sh"):
```

Comment:
this might be easier: `-DCMAKE_SH="CMAKE_SH-NOTFOUND"`

Reply:
Interesting, but I can't find this in the CMake docs. Do you have some link? From which cmake version does this work?

Reply:
in the past, I have found it here: https://stackoverflow.com/questions/41333215/mingw-sh-exe-must-not-be-in-your-path
it adds a cache entry for the `CMAKE_SH`, which is going to be used by `find_program(sh)`.
reading the [doc](https://cmake.org/cmake/help/latest/command/find_program.html), there are few parts to make it work:
```
If nothing is found, the result will be <VAR>-NOTFOUND
```

```
If nothing is found, the result will be <VAR>-NOTFOUND
```

Reply:
Created a new issue from this, can close it here now